ADR/0045-guidelines-for-developing-tasks.md

```diff
@@ -0,0 +1,54 @@
+# 45. Guidelines for developing tasks
+
+* Date: 2024-11-06
+
+## Status
+
+Proposed
+
+## Context
+
+One of the architectural goals for Konflux is extensibility. The core system provides opinionated [build pipelines]
+and [release pipelines], but lets users extend those and/or create their own. Beyond this goal, however, there has
+not been any clear guidance about where contributions should be added or the types of functionality that are typical
+for the stages of software development.
+
+## Decision
+
+* If any process is required to modify artifacts themselves, it should be done in [build pipelines]. This will ensure
```

Comment:
I'm not aware any use cases that modify the artifacts from the build pipeline. Perhaps a footnote/appendix with examples of such modifications could be useful. Some OLM use cases I'm not aware of?

This could be even broader/stricter, e.g. "any production of the artifact must be performed on the build pipelines with the intent that the produced artifact is shipped as-is".

Reply:
https://github.com/konflux-ci/release-service-catalog/pull/656 modifies the sbom, which is an artifact from the build pipeline